lib/wallaby/experimental/chrome.ex

```diff
@@ -87,64 +89,85 @@ defmodule Wallaby.Experimental.Chrome do
   end
 
   def get_window_size(%Session{} = session) do
-    handle = WebdriverClient.window_handle(session)
-    WebdriverClient.get_window_size(session, handle)
+    handle = delegate(:window_handle, session)
+    delegate(:get_window_size, session, [handle])
   end
 
   def set_window_size(session, width, height) do
-    handle = WebdriverClient.window_handle(session)
-    WebdriverClient.set_window_size(session, handle, width, height)
+    handle = delegate(:window_handle, session)
+    delegate(:set_window_size, session, [handle, width, height])
   end
 
   def accept_dialogs(_session), do: {:error, :not_implemented}
   def dismiss_dialogs(_session), do: {:error, :not_implemented}
-  
-  defdelegate accept_alert(session, fun),                        to: WebdriverClient
-  defdelegate dismiss_alert(session, fun),                       to: WebdriverClient
-  defdelegate accept_confirm(session, fun),                      to: WebdriverClient  
-  defdelegate dismiss_confirm(session, fun),                     to: WebdriverClient
-  defdelegate accept_prompt(session, input, fun),                to: WebdriverClient
-  defdelegate dismiss_prompt(session, fun),                      to: WebdriverClient
+
+  defp delegate(fun, element_or_session, args \\ []) do
+    check_logs!(element_or_session, fn ->
+      apply(WebdriverClient, fun, [element_or_session | args])
+    end)
+  end
+
+  defdelegate accept_alert(session, fun),         to: WebdriverClient
+  defdelegate dismiss_alert(session, fun),        to: WebdriverClient
+  defdelegate accept_confirm(session, fun),       to: WebdriverClient
+  defdelegate dismiss_confirm(session, fun),      to: WebdriverClient
+  defdelegate accept_prompt(session, input, fun), to: WebdriverClient
+  defdelegate dismiss_prompt(session, fun),       to: WebdriverClient
+  defdelegate parse_log(log),                     to: Wallaby.Experimental.Chrome.Logger
 
 
   @doc false
-  defdelegate cookies(session),                                   to: WebdriverClient
+  def cookies(session), do: delegate(:cookies, session)
   @doc false
-  defdelegate current_path(session),                              to: WebdriverClient
+  def current_path(session), do: delegate(:current_path, session)
   @doc false
-  defdelegate current_url(session),                               to: WebdriverClient
+  def current_url(session), do: delegate(:current_url, session)
   @doc false
-  defdelegate page_title(session),                                to: WebdriverClient
+  def page_title(session), do: delegate(:page_title, session)
   @doc false
-  defdelegate page_source(session),                               to: WebdriverClient
+  def page_source(session), do: delegate(:page_source, session)
   @doc false
-  defdelegate set_cookie(session, key, value),                    to: WebdriverClient
+  def set_cookie(session, key, value), do: delegate(:set_cookie, session, [key, value])
```

Comment:
I know this doesn't affect our public API, but I keep feeling like the `delegate` calls make it difficult to figure out what's going on at a glance. Maybe there's a better name for this function, or maybe it would be worth just writing it out.
```elixir
def set_cookie(session, key, value) do
  check_logs!(session, fn ->
    WebdriverClient.set_cookies(session, key, value)
  end)
end
```
Do you think there is a way to make it more clear what's going on? @keathley, @PragTob what do you think? Is this worth addressing before merging the PR?

Reply:
I'm not sure. Writing it out is certainly more explicit. Honestly, I think all of the interactions between the different driver modules and the httpclient are in need of some refactoring. My feeling is that this is alright for now and we can work on refactoring all of these apis as a separate task.  A simple fix for now might be to simple rename `delegate/3` to `webdriver/3` or `httpclient/3`?

Reply:
You bring up a good point, @keathley. If we're going to be refactoring the internals of these main driver modules anyways, there's no need worry about the naming of this private function. @chriserin Don't worry about making this change.